src/meta/src/stream/source_manager.rs

```diff
@@ -1136,15 +1141,28 @@ impl SourceManager {
     /// The command will first updates `SourceExecutor`'s splits, and finally calls `Self::apply_source_change`
     /// to update states in `SourceManager`.
     async fn tick(&self) -> MetaResult<()> {
-        let split_assignment = {
+        let mut split_assignment = {
             let core_guard = self.core.lock().await;
             core_guard.reassign_splits().await?
         };
 
+        let dup_assignment_flag = split_assignment
+            .iter_mut()
+            .map(|(_, assignment)| validate_assignment(assignment))
+            .reduce(|a, b| a && b)
+            .unwrap_or(false);
+
         if !split_assignment.is_empty() {
             let command = Command::SourceSplitAssignment(split_assignment);
             tracing::info!(command = ?command, "pushing down split assignment command");
-            self.barrier_scheduler.run_command(command).await?;
+            if dup_assignment_flag {
+                tracing::warn!("duplicate split assignment found, wrap with pause and resume");
+                self.barrier_scheduler
+                    .run_config_change_command_with_pause(command)
```

Comment:
> This pr handles this case by wrapping the dup split detected split_change_mutation with pause/resume. Separating the two writes to different epoches.

If the goal is "separating the two writes to different epoches", why use pause/resume instead of two ordinary barriers?

Reply:
To achieve exactly-once promise. After actor_1 writes split_2's latest offset, there should be no new data come in before actor_2 starts reading split_2. 
The case above is actually split migration, it also happens when cluster scales, and we treat such case using Update mutation, wrapped by pause/resume.